Sources/AppHost/OAAppDelegate.mm

```diff
@@ -175,90 +177,267 @@ - (BOOL) initialize
+    }
+    return NO;
+}
+
+- (BOOL)handleIncomingMoveMapToLocationURL:(NSURL *)url
+{
+    NSString *pathPrefix = @"/map#";
+    NSInteger pathStartIndex = [url.absoluteString indexOf:pathPrefix];
+    if (_rootViewController && pathStartIndex != -1 && [OAUtilities isOsmAndMapUrl:url])
+    {
+        NSArray<NSString *> *params = [[url.absoluteString substringFromIndex:pathStartIndex + pathPrefix.length] componentsSeparatedByString:@"/"];
+        if (params.count == 3)
+        {
+            int zoom = [params[0] intValue];
+            double lat = [params[1] doubleValue];
+            double lon = [params[2] doubleValue];
+            [self moveMapToLat:lat lon:lon zoom:zoom withTitle:nil];
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (BOOL)handleIncomingOpenLocationMenuURL:(NSURL *)url
+{
+    if ([OAUtilities isOsmAndGoUrl:url])
+    {
+        NSURLComponents *components = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:YES];
+        NSArray<NSURLQueryItem *> *queryItems = components.queryItems;
+        NSString *latParam;
+        NSString *lonParam;
+        NSString *zoomParam;
+        NSString *titleParam;
+        for (NSURLQueryItem *queryItem in queryItems)
+        {
+            if ([queryItem.name.lowercaseString isEqualToString:@"lat"])
+                latParam = queryItem.value;
+            else if ([queryItem.name.lowercaseString isEqualToString:@"lon"])
+                lonParam = queryItem.value;
+            else if ([queryItem.name.lowercaseString isEqualToString:@"z"])
+                zoomParam = queryItem.value;
+            else if ([queryItem.name.lowercaseString isEqualToString:@"title"])
+                titleParam = queryItem.value;
+        }
+
+        if (_rootViewController && latParam && lonParam)
+        {
+            double lat = [latParam doubleValue];
+            double lon = [lonParam doubleValue];
+            int zoom = _rootViewController.mapPanel.mapViewController.mapView.zoom;
+            if (zoomParam)
+                zoom = [zoomParam intValue];
+            [self moveMapToLat:lat lon:lon zoom:zoom withTitle:titleParam];
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (BOOL)handleIncomingTileSourceURL:(NSURL *)url
+{
+    if (_rootViewController && [OAUtilities isOsmAndSite:url] && [OAUtilities isPathPrefix:url pathPrefix:@ "/add-tile-source"])
+    {
+        NSDictionary<NSString *, NSString *> *params = [OAUtilities parseUrlQuery:url];
+        // https://osmand.net/add-tile-source?name=&url_template=&min_zoom=&max_zoom=
+        OAOnlineTilesEditingViewController *editTileSourceController = [[OAOnlineTilesEditingViewController alloc] initWithUrlParameters:params];
+        [_rootViewController.navigationController pushViewController:editTileSourceController animated:NO];
         return YES;
     }
-        
+    return NO;
+}
+
+- (void)buildRoute:(CLLocation *)start end:(CLLocation *)end appMode:(OAApplicationMode *)appMode
```

Comment:
Not the best place for this and the next method, forward the parameters to a more appropriate class and handle it there (for example in the rootViewController or MapPanel